packages/story-editor/src/app/media/local/useContextValueProvider.js

```diff
@@ -210,19 +223,10 @@ export default function useContextValueProvider(reducerState, reducerActions) {
 
   const postProcessingResource = useCallback(
     (resource) => {
-      const {
-        local,
-        type,
-        isMuted,
-        baseColor,
-        src,
-        id,
-        posterId,
-        mimeType,
-        poster,
-      } = resource;
+      const { type, isMuted, baseColor, src, id, posterId, mimeType, poster } =
+        resource;
 
-      if (local || !id) {
+      if (!id || isResourceUploadingById(id)) {
```

Comment:
Id is a required field, as without an id ( like hotlinked images ) then most of the code below would fail.

Reply:
I know that `id` is required.

My point is: all resources are always guaranteed to have an ID. (and if not, we should fix that, as mentioned on the ticket #9577)

Because of that, there is no need for these `id` checks everywhere.

Reply:
And yes, hotlinked images _do_ have an ID:

https://github.com/google/web-stories-wp/blob/f22db1e7ddc307a07fabc020c536eb6f8d8a8904/packages/story-editor/src/components/library/panes/media/local/hotlink/useInsert.js#L98-L104